src/components/TableComponent/index.jsx

```diff
@@ -1,166 +1,120 @@
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
 
-import { Pagination, Table } from '@edx/paragon';
+import { Pagination, DataTable, Alert } from '@edx/paragon';
+import { Error, WarningFilled } from '@edx/paragon/icons';
 import 'font-awesome/css/font-awesome.css';
 
 import TableLoadingSkeleton from './TableLoadingSkeleton';
-import TableLoadingOverlay from '../TableLoadingOverlay';
-import StatusAlert from '../StatusAlert';
 import { updateUrl } from '../../utils';
 
-class TableComponent extends React.Component {
-  componentDidMount() {
-    // Get initial data
-    this.props.paginateTable();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { location } = this.props;
-
-    // Handle the case where the query params have changed. This is used when sorting & paging, but
-    // also when the back button is used. We need to determine if this is a pagination or sorting
-    // request as we handle these as slightly different actions in the action handlers.
-    if (location.search !== prevProps.location.search) {
-      const prevQueryParams = new URLSearchParams(prevProps.location.search);
-      const prevPage = prevQueryParams.get('page');
-      const prevOrdering = prevQueryParams.get('ordering');
-      const currentQueryParams = new URLSearchParams(location.search);
-      const page = currentQueryParams.get('page');
-      const ordering = currentQueryParams.get('ordering');
-      if (ordering !== prevOrdering) {
-        this.props.sortTable(ordering);
-      } else if (page !== prevPage) {
-        this.props.paginateTable(parseInt(page, 10));
-      }
-    }
-  }
-
-  componentWillUnmount() {
-    this.props.clearTable();
-  }
-
-  renderTableContent() {
-    const {
-      className,
-      currentPage,
-      pageCount,
-      tableSortable,
-      data,
-      ordering,
-      formatData,
-      id,
-      loading,
-      enterpriseId,
-    } = this.props;
-
-    const sortByColumn = (column, direction) => {
+const TableComponent = ({
+  className, currentPage, pageCount, tableSortable, data, formatData, id, loading, enterpriseId, totalCount,
+  paginateTable, clearTable, columns, error, sortTable,
+}) => {
+  const sortByColumn = useCallback(args => {
+    const { sortBy } = args;
+    if (sortBy && sortBy.length > 0) {
+      const column = sortBy[0];
+      const ordering = `${column.desc ? '-' : ''}${column.id}`;
       updateUrl({
         page: 1,
-        ordering: direction === 'desc' ? `-${column.key}` : column.key,
+        ordering,
       });
+      sortTable(ordering);
```

Comment:
Ideally, our tables throughout the app would no longer need to dispatch Redux actions to do pagination/sorting/filtering, etc. but instead use what `DataTable` offers you via `fetchData` directly.